plugin/src/integrationTest/kotlin/com/vanniktech/maven/publish/ProjectSpecs.kt

```diff
@@ -0,0 +1,119 @@
+  "POM_INCEPTION_YEAR" to "2018",
+  "POM_URL" to "https://github.com/vanniktech/gradle-maven-publish-plugin/",
+
+  "POM_SCM_URL" to "https://github.com/vanniktech/gradle-maven-publish-plugin/",
+  "POM_SCM_CONNECTION" to "scm:git:git://github.com/vanniktech/gradle-maven-publish-plugin.git",
+  "POM_SCM_DEV_CONNECTION" to "scm:git:ssh://git@github.com/vanniktech/gradle-maven-publish-plugin.git",
+
+  "POM_LICENCE_NAME" to "The Apache Software License, Version 2.0",
+  "POM_LICENCE_URL" to "https://www.apache.org/licenses/LICENSE-2.0.txt",
+  "POM_LICENCE_DIST" to "repo",
+
+  "POM_DEVELOPER_ID" to "vanniktech",
+  "POM_DEVELOPER_NAME" to "Niklas Baudy",
+  "POM_DEVELOPER_URL" to "https://github.com/vanniktech/",
+)
+
+fun javaProjectSpec() = ProjectSpec(
+  plugins = listOf(
+    javaPlugin,
+  ),
+  group = "com.example",
+  artifactId = "test-artifact",
+  version = "1.0.0",
+  properties = defaultProperties,
+  sourceFiles = listOf("src/main/java" to "com/vanniktech/maven/publish/test/JavaTestClass.java")
+)
+
+fun javaLibraryProjectSpec() = ProjectSpec(
+  plugins = listOf(
+    javaLibraryPlugin,
+  ),
+  group = "com.example",
+  artifactId = "test-artifact",
+  version = "1.0.0",
+  properties = defaultProperties,
+  sourceFiles = listOf("src/main/java" to "com/vanniktech/maven/publish/test/JavaTestClass.java")
+)
+
+fun kotlinJvmProjectSpec(version: KotlinVersion) = ProjectSpec(
+  plugins = listOf(
+    kotlinJvmPlugin.copy(version = version.value)
+  ),
+  group = "com.example",
+  artifactId = "test-artifact",
+  version = "1.0.0",
+  properties = defaultProperties,
+  sourceFiles = listOf(
+    "src/main/java" to "com/vanniktech/maven/publish/test/JavaTestClass.java",
+    "src/main/kotlin" to "com/vanniktech/maven/publish/test/KotlinTestClass.kt",
+  )
+)
+
+fun kotlinJsProjectSpec(version: KotlinVersion) = ProjectSpec(
+  plugins = listOf(
+    kotlinJsPlugin.copy(version = version.value)
+  ),
+  group = "com.example",
+  artifactId = "test-artifact",
+  version = "1.0.0",
+  properties = defaultProperties,
+  sourceFiles = listOf(
+    "src/main/kotlin" to "com/vanniktech/maven/publish/test/KotlinTestClass.kt",
+  ),
+  buildFileExtra = """
+    kotlin {
+        js {
+            nodejs()
+        }
+    }
+  """.trimIndent()
+)
+
+fun androidLibraryProjectSpec(version: AgpVersion) = ProjectSpec(
+  plugins = listOf(
+    androidLibraryPlugin.copy(version = version.value)
+  ),
+  group = "com.example",
+  artifactId = "test-artifact",
+  version = "1.0.0",
+  properties = defaultProperties,
+  sourceFiles = listOf(
+    "src/main/java" to "com/vanniktech/maven/publish/test/JavaTestClass.java",
+  ),
+  buildFileExtra = """
+    android {
+        namespace "com.test.library"
+        compileSdkVersion 29
+    }
+
+    // disable the dokka task to speed up Android tests significantly
+    afterEvaluate {
+      tasks.named("javaDocReleaseGeneration").configure {
+        it.enabled = false
+      }
+    }
```

Comment:
This isn't great but it will still produce an empty javadoc jar and our tests are only checking for existence of them and never look at the content.

Disabled task:
![Screenshot 2022-10-21 at 14 55 35](https://user-images.githubusercontent.com/1358105/197209450-d1e9d79b-01d4-4bda-bd64-cfc3adb47a29.png)
Enabled task:
![Screenshot 2022-10-21 at 15 37 40](https://user-images.githubusercontent.com/1358105/197209456-e046ae60-bc21-451c-8c43-8ce65e8191f9.png)